### ISSUE TITLE
[Bug]: WARNING:autogen.io.base:No default IOStream has been set, defaulting to IOConsole.

### ISSUE THREAD

**Issue by daniel-counto** [microsoft/autogen#2204]
### Describe the bug

This seems like a code update from Autogen and/or gpt4-v, and it breaks my code.
I use gpt4-v (vision) for the agents, now it is not able to read images that it was able to read yesterday. Now it gives the errors: 
"WARNING:autogen.io.base:No default IOStream has been set, defaulting to IOConsole."
and
"Warning! Unable to load image from /content/drive/MyDrive/some_image.jpg, because Invalid base64-encoded string: number of data characters (53) cannot be 1 more than a multiple of 4"

### Steps to reproduce

how to replicate? just rerun any application with gpt4-v now and it will appear.
I am using Google Colab.

### Model Used

gpt4-v

### Expected Behavior

suppose to read the image without any issue

### Screenshots and logs

<img width="1214" alt="Screenshot 2024-03-29 at 3 47 35 PM" src="https://github.com/microsoft/autogen/assets/140989990/14927fd6-37b5-442f-8afe-650e9a978cde">

**Comment by Einengutenmorgen** [comment]
SAME

**Comment by davorrunje** [comment]
Unless you are using Websockets, the `IOStream` warning can be ignored.

**Comment by ekzhu** [comment]
@BeibinLi @WaelKarkoub please help. With the GPT-V issue.

**Comment by WaelKarkoub** [comment]
Hi @daniel-counto, are you able to run this notebook https://github.com/microsoft/autogen/blob/main/notebook/agentchat_lmm_gpt-4v.ipynb? I wasn't able to reproduce your error

**Comment by daniel-counto** [comment]
Thanks guys, it is weird that the same image file is causing this error today. However, I convert the original pdf file to image again and now the new image is able to be read without any issues. Seem's like not a gpt4-v issue.

**Comment by jansteh** [comment]
How to se default to IOConsole? (so I dont see the IOstream warning?)
Or can you please point me to documentation?

**Comment by wanghaoxue0** [comment]
I have the same issue

**Comment by daoxuliu** [comment]
I have the same issue too

**Comment by Daniel-Northcutt** [comment]
Running in Databricks notebook and having this issue since the most recent release. Present in chat messages and function calls.

**Comment by GitDakky** [comment]
I only ever see this while using AutoGen Studio - Don't get it in terminal

**Comment by davorrunje** [comment]
This warning messages has been removed as there is no danger of anything breaking. You can ignore it in this version and it will go away in the next one anyway.

**Comment by ctavolazzi** [comment]
I'm also experiencing the error, but it's different.

The beginning of the error is the same:

> python app.py
No default IOStream has been set, defaulting to IOConsole.
No default IOStream has been set, defaulting to IOConsole.
No default IOStream has been set, defaulting to IOConsole.
UserProxy (to Assistant):

At the end I'm getting:

  File "C:\Users\ctavo\Miniconda3\envs\autogen\Lib\site-packages\openai\_utils\_utils.py", line 275, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
TypeError: Completions.create() got an unexpected keyword argument 'request_timeout'

**Comment by WaelKarkoub** [comment]
@ctavolazzi how did you setup your LLM config?

**Comment by davorrunje** [comment]
@ctavolazzi The warnings about IOStream can be safely ignored.

**Comment by Zhuoqun123** [comment]
I see the same warning information as 
--------------------------------------------------------------------------------
No default IOStream has been set, defaulting to IOConsole.
No default IOStream has been set, defaulting to IOConsole.
No default IOStream has been set, defaulting to IOConsole.
User_proxy (to Pattern_Agent):

although they are not influence the result, this error appear too much time in my terminal, I hope to know how to remove it?

**Comment by davorrunje** [comment]
> ## I see the same warning information as
> No default IOStream has been set, defaulting to IOConsole. No default IOStream has been set, defaulting to IOConsole. No default IOStream has been set, defaulting to IOConsole. User_proxy (to Pattern_Agent):
> 
> although they are not influence the result, this error appear too much time in my terminal, I hope to know how to remove it?

@Zhuoqun123 This is already removed in the main branch, you can install it from git instead from the pypi.

**Comment by Princekrampah** [comment]
Same issue as well

**Comment by jackgerrits** [comment]
@Princekrampah this has been fixed in main, just waiting on a release

**Comment by aullrich2013** [comment]
FYI I've used the following approach to work around the issue temporarily. 

`class fakeIOStream():
    def __init__(self):
        pass
    def print(self, *args, **kwargs):
        print(*args, **kwargs)
    def input(self, *args, **kwargs):
        return input(*args, **kwargs)
    

autogen.io.IOStream.set_global_default(fakeIOStream())
autogen.io.IOStream.set_default(fakeIOStream())
autogen.io.IOStream._default_io_stream.set(fakeIOStream())`

**Comment by davorrunje** [comment]
> FYI I've used the following approach to work around the issue temporarily.
> 
> ```
> class fakeIOStream():
>     def __init__(self):
>         pass
>     def print(self, *args, **kwargs):
>         print(*args, **kwargs)
>     def input(self, *args, **kwargs):
>         return input(*args, **kwargs)
>     
> 
> autogen.io.IOStream.set_global_default(fakeIOStream())
> autogen.io.IOStream.set_default(fakeIOStream())
> autogen.io.IOStream._default_io_stream.set(fakeIOStream())```
> ```

This will not work in some (probably most) situations. Whenever you create a new thread, the default output stream should be set up. If it is not set, it will default to IOConsole and continue as is. This is the common pattern in distributed settings where you might have a single agent serving multiple users over websockets. If you are not using websockets, you can just ignore the warning as it is remove from the latest version anyway.

**Comment by sschet** [comment]
I was using the below code for this error on 4/4:

```
@staticmethod
def set_global_default(stream: "IOStream") -> None
```

but now that code results in the below error:

```
Cell In[23], line 12
    def set_global_default(stream: "IOStream") -> None
                                                                                      ^
SyntaxError: expected ':'
```

**Comment by ekzhu** [comment]
> I was using the below code for this error on 4/4/2024:
> 
> ```
> @staticmethod
> def set_global_default(stream: "IOStream") -> None
> ```
> 
> but today (4/8/2024) that code results in the below error:
> 
> ```
> Cell In[23], line 12
>     def set_global_default(stream: "IOStream") -> None
>                                                                                       ^
> SyntaxError: expected ':'
> ```

Is it a bug or a question? For a bug can you create a separate issue? For question can you create a Q&A in our GitHub Discussion?
